tensorflow/lite/micro/micro_time.h

```diff
@@ -26,9 +26,9 @@ int32_t ticks_per_second();
 // Return time in ticks.  The meaning of a tick varies per platform.
 int32_t GetCurrentTimeTicks();
 
-inline int32_t TicksToMs(int32_t ticks) {
-  return static_cast<int32_t>(1000.0f * static_cast<float>(ticks) /
-                              static_cast<float>(ticks_per_second()));
+inline uint32_t TicksToMs(uint32_t ticks) {
```

Comment:
Did you run into some trouble with int32? I don't think this should be necessary.

The duration calculation https://github.com/tensorflow/tensorflow/blob/996f7d5041133e71f4a9b2f65e87b404a83b8007/tensorflow/lite/micro/micro_profiler.cc#L52

will have the same range whether we have ticks be int32 or uint32. The only requirement is that no more than 2**32 ticks should have passed between two calls to BeginEvent/EndEvent, which I believe has a safe margin.

And there isn't any reason for TicksToMs to return a uint32 since int32 gives us a range of 2147483 seconds.

Reply:
Yes I ran into problems.. need to change from int32_t ticks to uint32_t ticks in PersonDetectionNIerations() in person_detection_benchmark.cc, when running make -j -f tensorflow/lite/micro/tools/make/Makefile TARGET=cortex_m_corstone_300 TARGET_ARCH=cortex-m55 run_person_detection_benchmark - otherwise getting negative results. Then I figure why not change to uint32_t as well in TicksToMs() but you are right TicksToMs() can still use int32_t. I will change back..

Reply:
ok, I saw the change that you made to change accumulation to uint32, and that makes sense.